### PR TITLE
(just for my self-learning) add paddle.nn.functional.pairwise_distance

### PR DESCRIPTION
**Pull request by OccupyMars2030**
### PR types

New features 
### PR changes

APIs 
### Describe

add paddle.nn.functional.pairwise_distance
(1. move the code of the forward method of  paddle.nn.PairwiseDistance  to paddle.nn.functional.pairwise_distance  
2. just use paddle.nn.functional.pairwise_distance  in  the forward method of  paddle.nn.PairwiseDistance   )

### CODE REVIEW

**Comment by OccupyMars2030** [comment]
Problem 1:
`    if in_dygraph_mode():
        sub = _C_ops.elementwise_sub(x, y)
        # p_norm op has not uesd epsilon, so change it to the following.
        if epsilon != 0.0:
            epsilon = paddle.fluid.dygraph.base.to_variable([epsilon],
                                                            dtype=sub.dtype)
            sub = _C_ops.elementwise_add(sub, epsilon)
        return _C_ops.final_state_p_norm(sub, p, -1, 0., keepdim, False)

    if _in_legacy_dygraph():
        sub = _C_ops.elementwise_sub(x, y)
        if epsilon != 0.0:
            epsilon = paddle.fluid.dygraph.base.to_variable([epsilon],
                                                            dtype=sub.dtype)
            sub = _C_ops.elementwise_add(sub, epsilon)
        return _C_ops.p_norm(sub, 'axis', -1, 'porder', p, 'keepdim', keepdim,
                             'epsilon', 0.)`

if " p_norm op has not uesd epsilon" , then  is it necessary  to  modify  p_norm  op  to  make  it  use  
'epsilon'  argument

**Comment by OccupyMars2030** [comment]
`import paddle
import numpy as np
paddle.disable_static()
x_np = np.array([[1., 3.], [3., 5.], [4, 4]]).astype(np.float64)
# y_np = np.array([[5., 6.], [7., 8.], [4, 3]]).astype(np.float64)
x = paddle.to_tensor(x_np)
# y = paddle.to_tensor(y_np)
dist = paddle.nn.PairwiseDistance(p=2, keepdim=True)
distance = dist(x, x)
print(distance) `


run the above code, generate all-zero  tensor,   what's  the point  of  'epsilon'  argument ???

**Comment by OccupyMars2030** [comment]
`print(np.linalg.norm(np.zeros((2, 3)), ord=2, axis=1, keepdims=True))
`

no 'epsilon'  argument !!!

**Comment by OccupyMars2030** [comment]
`x - y + epsilon`   is  not  convincing,   maybe  it  should  be changed  to  `|x - y| + epsilon`

**Comment by OccupyMars2030** [comment]
Can we  expose  the  'axis'  argument  in the pairwise_distance  API  because  'axis'  argument  exits in the  p_norm  op   API  ?

**Comment by OccupyMars2030** [comment]
import paddle
import numpy as np
paddle.disable_static()
x_np = np.array([[1., 3.], [3., 5.], [4, 4]]).astype(np.float64)
y_np = np.array([[5., 6.], [7., 8.], [4, 3]]).astype(np.float64)
x = paddle.to_tensor(x_np)
y = paddle.to_tensor(y_np)
dist = paddle.nn.PairwiseDistance(p=2, epsilon=1000, keepdim=True)
distance = dist(x, y)
print(distance.numpy())

distance_2 = np.linalg.norm(x - y, ord=2, axis=-1, keepdims=True)
print(distance_2)


The above code shows that 'epsilon' is useless
